### ISSUE TITLE
Excessive Screen Reader Feedback while typing in the integrated terminal

### ISSUE THREAD

**Issue by rperez030** [microsoft/vscode#243466]
Type: <b>Bug</b>

When using the terminal in VSCode with screen reader support, the terminal input field mistakenly triggers live updates. This issue results in the screen reader reading the entire content of the text field twice with each keypress, which is not the intended behavior. I believe the terminal input field may have been accidentally included in an aria-live region.

## Steps to Reproduce
1. Open VSCode and navigate to the terminal.
2. Ensure your screen reader is active but has the key character echo and word echo features turned off.
3. Type a phrase in the terminal, such as "this is a test."
4. Observe that after each keypress, the full content of the terminal input field is redundantly read by the screen reader.

## Expected Behavior
When typing in the terminal with keyboard echo turned off, the screen reader should remain silent, preventing disruptive and unnecessary feedback.

## Actual Behavior
Each keystroke results in the entire terminal input being read aloud, even with key echo features turned off,.

CC @meganrogge 

VS Code version: Code - Insiders 1.99.0-insider (b59ae26ad3882edfec6493e280784f5395978dbd, 2025-03-13T09:44:45.504Z)
OS version: Windows_NT x64 10.0.26100
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10610U CPU @ 1.80GHz (8 x 2304)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|47.76GB (32.59GB free)|
|Process Argv|. --enable-proposed-api msft-1es.es-chat --crash-reporter-id bc5798b9-59a5-4bd3-88aa-596610b38a0f|
|Screen Reader|no|
|VM|40%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt551cf:31249598
vscod805cf:30301675
py29gd2263:31024238
14424t2-chatv7:31259162
c4g48928:30535728
962ge761:30841072
dsvsc014:30777825
dsvsc015:30821418
2e7ec940:31000449
pythontbext0:30879054
cppperfnew:30980852
pythonait:30973460
dwnewjupyter:31046869
nativerepl1:31134653
pythonrstrctxt:31093868
nativeloc1:31118317
e80f6927:31120813
iacca1:31150324
notype1:31143044
dwcopilot:31158714
c3hdf307:31184662
6074i472:31201624
dwoutputs:31242946
customenabled:31248079
hdaa2157:31222309
copilot_t_ci:31222730
f27dg485:31233432
g012b348:31231168
jda6j935:31233686
copilothoveron:31247701
pythoneinst12:31251391
treesitterini:31258269
bgtreat:31252123
fh1c7952:31247580
5e5g2996:31258849
hb9j0513:31256344
pythonpulldiagcf:31258011

```

</details>

**Comment by meganrogge** [comment]
Hi @rperez030 thank you for the issue. Can you reproduce this in VS Code stable?

**Comment by meganrogge** [comment]
here's a recording with sound that follows your steps on macos with voice over and i'm not reproducing the issue. 

> but has the key character echo and word echo features turned off.

How do I do that? will try now

https://github.com/user-attachments/assets/f67eeab2-a3e1-4814-8c91-e06632d792df

**Comment by meganrogge** [comment]
In VoiceOver settings, I just changed the text setting to while typing speak... to nothing. I still am not seeing the issue you're describing. When i do that, I don't hear anything unless I use arrow keys.

**Comment by rperez030** [comment]
> Hi [@rperez030](https://github.com/rperez030) thank you for the issue. Can you reproduce this in VS Code stable?

Good question. I cannot reproduce it in the stable version. I can reproduce it using macOS VoiceOver, but it is not as consistent as it is on Windows. I have to type very slowly, and it only seems to fail for the first few letters.

To adjust verbosity while typing in VoiceOver, open the VoiceOver Utility and navigate to Verbosity / Text. Under "While Typing Speak," select "Nothing." Alternatively, press VO + V anywhere in the system, press the right arrow until you hear "Typing Echo," and use the up/down arrows to select "Nothing."

**Comment by meganrogge** [comment]
Got it - do you have any terminal settings? Are you hearing anything weird in the editor?

**Comment by meganrogge** [comment]
In both stable and Insider's, using VoiceOver with the `While typing speak: Do Nothing`, I hear the same - no chars are announced unless I navigate with arrow keys. I can also reproduce that in the default mac terminal.

I do not hear live announcements while typing though. 

May I ask why you've adjusted that setting? Is it usually set to `Do nothing` for you?

**Comment by meganrogge** [comment]
Maybe a screen recording would help?

**Comment by rperez030** [comment]
> May I ask why you've adjusted that setting? Is it usually set to `Do nothing` for you?

Right. I need silence while typing, otherwise I get very distracted. I did some debugging with NVDA on Windows, and there is definitely a live region changed event that fires for every key press.

`Hook received winEvent: EVENT_OBJECT_LIVEREGIONCHANGED, window 1772494 (Chrome_RenderWidgetHostHWND), objectID OBJID_CLIENT, childID -1873, process 1264 (code - insiders), thread 24332`

screen recording coming up.

**Comment by meganrogge** [comment]
Got it and thanks - do you have any settings related to the terminal?

**Comment by rperez030** [comment]
I'm not aware of any setting, but I can try creating an empty profile to double-check. like I say in the recording, this is particularly disruptive when using GitHub Copilot in agent mode because when the agent enters a command in the terminal, it creates and endless string of text.

https://github.com/user-attachments/assets/3e6e6664-37cd-4bbc-a7f6-80ce55e317b9

**Comment by Tyriar** [comment]
@meganrogge this is the code that handles these events which hasn't changed in a while:

https://github.com/xtermjs/xterm.js/blob/2042bb85023714e55c0c2e986b5000e33b17c414/src/browser/AccessibilityManager.ts#L41-L50

The `charsToConsume` part suppresses incoming input later on:

https://github.com/xtermjs/xterm.js/blob/2042bb85023714e55c0c2e986b5000e33b17c414/src/browser/AccessibilityManager.ts#L141-L149

It's more complicated here because we have the input event (that a screen reader may echo), and then the asynchronous "draw" to the terminal later on. So it doesn't work like a lot of web content.

---

The problem here looks like it's PSReadLine being activated. I actually spotted this recently, I think the issue is here:

https://github.com/microsoft/vscode/blob/38d1f292632321d5a256a4f66480625a1e18e6d8/src/vs/workbench/contrib/terminal/common/scripts/shellIntegration.ps1#L165

We need that to be `if (Get-Module -Name PSReadLine) {` to not implicitly load the PSReadLine module.

**Comment by amunger** [comment]
still seems to repro for me in pwsh

![Image](https://github.com/user-attachments/assets/68154e09-62b2-4b05-a4f0-de3120e7bfec)

**Comment by Tyriar** [comment]
@amunger do you have screen reader on and not touch any readline APIs like `Get-PSReadLineOption` in your profile? If you do the latter it's expected that this would still happen since that implicitly imports `PSReadLine`.

**Comment by meganrogge** [comment]
@rperez030 would be good if you could verify this is fixed.

**Comment by meganrogge** [comment]
@rperez030 said this works like a charm now! great work @tyriar!

![image of chat with Roberto that says this works like a charm in the terminal](https://github.com/user-attachments/assets/f52abda8-8e85-47a0-9b8c-54bbb17232d7)

**Comment by rperez030** [comment]
Yes. confirmed in both mac and Windows with the pwsh profile on windows. Thanks everyone for such a quick turnaround.
